repose-aggregator/components/services/atom-feed-service/atom-feed-service-api/src/main/java/org/openrepose/nodeservice/atomfeed/AuthenticatedRequestFactory.java

```diff
@@ -29,15 +27,24 @@
     /**
      * Mutates a request to adhere to some authentication scheme compatible with the Atom service.
      *
-     * @param atomFeedUrlConnection The unauthenticated feed {@link URLConnection} with the Atom service.
-     * @param context               A context object which contains information related to the request.
-     * @return The {@link URLConnection} with authentication mutations applied.
+     * @param feedReadRequest A mutable container for request data that will be sent to the Atom service.
+     * @param context         A context object which contains information related to the request.
+     * @return The {@link FeedReadRequest} with authentication mutations applied.
+     * @throws AuthenticationRequestException if the request fails to authenticate.
      */
-    URLConnection authenticateRequest(URLConnection atomFeedUrlConnection, AuthenticationRequestContext context);
+    FeedReadRequest authenticateRequest(FeedReadRequest feedReadRequest, AuthenticationRequestContext context) throws AuthenticationRequestException;
 
     /**
      * This method will be called anytime authentication credentials on a request that has been processed by the
      * authenticateRequest method are found to be invalid.
      */
     void onInvalidCredentials();
+
+    /**
+     * Sets the configured Connection Pool Id.
+     * <p>
+     * NOTE: This default implementation does nothing and will need to be overridden if the value is required.
+     */
+    default void setConnectionPoolId(String poolId) {
```

Comment:
this isn't meant to be part of the interface for this, because not all implementations are going to need to make requests. The intent was to wire the feed's configuration into the auth component. Not the entire feed service config, just the feed itself.